src/Data/Vec/Functional/Relation/Unary/All.agda

```diff
@@ -0,0 +1,126 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- All lifting of predicates to index notation vectors
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Vec.Functional.Relation.Unary.All where
+
+open import Data.Fin
+open import Data.Fin.Properties
+open import Data.Nat
+open import Data.Product as Σ using (_×_; _,_; proj₁; proj₂; uncurry)
+open import Data.Vec.Functional as VF hiding (map)
+open import Function
+open import Level using (Level)
+open import Relation.Unary
+
+private
+  variable
+    a b c p q r ℓ : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+------------------------------------------------------------------------
+-- Definition
+
+All : Pred A ℓ → ∀ {n} → Vector A n → Set ℓ
+All P xs = ∀ i → P (xs i)
+
+------------------------------------------------------------------------
+-- Operations
+
+map : {P : Pred A p} {Q : Pred A q} → P ⊆ Q → ∀ {n} → All P {n = n} ⊆ All Q
+map pq ps i = pq (ps i)
+
+------------------------------------------------------------------------
+-- map
+
+map⁺ : {P : Pred A p} {Q : Pred B q} {f : A → B} →
```

Comment:
The convention in other parts of the library has been to put all the `⁺` and `⁻` proofs in `All.Properties` rather than `All`.